### PR TITLE
[PM-1802] Update 2fa provider icons and description

### PR DESCRIPTION
**Pull request by quexten**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-1802

## 📔 Objective

Update two-factor login and management components (description and icons).

Considerations: I moved the SVG data into a new [two-factor-icon.component.html](https://github.com/bitwarden/clients/pull/9568/files#diff-d7ec25dcda16e16cbd5dd37a7fc1c79814305fac715f89bd0463c035bbf96069). Does it make sense to split this into further files, once for each svg?

## 📸 Screenshots

image
image
image
image
image
image

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: libs/angular/src/auth/components/two-factor-icon.component.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import { Component, Input } from "@angular/core";
+
+@Component({
+  selector: "bw-two-factor-icon",
```
Comment (rr-bw) [inline libs/angular/src/auth/components/two-factor-icon.component.ts:4]: Why `bw-` prefix? I think we'd just go with `auth-` here.

**Review by rr-bw — CHANGES_REQUESTED** [review]
Looks great so far. A couple of minor comments and copy changes below. As for splitting the SVGs into separate files, I think that's a good idea. You could try following the pattern [here](https://github.com/bitwarden/clients/blob/main/apps/web/src/app/auth/login/login-via-webauthn/login-via-webauthn.component.ts#L12) and [here](https://github.com/bitwarden/clients/blob/main/libs/angular/src/auth/icons/create-passkey.icon.ts) using `svgIcon`.

File: apps/browser/src/_locales/en/messages.json [hunk 2]
```diff
@@ -1093,17 +1093,17 @@
     "message": "Authenticator app"
   },
   "authenticatorAppDesc": {
-    "message": "Use an authenticator app (such as Authy or Google Authenticator) to generate time-based verification codes.",
-    "description": "'Authy' and 'Google Authenticator' are product names and should not be translated."
+    "message": "Enter a code generated by an authenticator app like Bitwarden Authenticator.",
+    "description": "'Bitwarden Authenticator' is a product name and should not be translated."
```
Comment (djsmith85) [inline apps/browser/src/_locales/en/messages.json:1097]: To let our translation community over at CrowdIn know, that the source string has changed, we also need to provide a new key. Please rename all keys where you have changed the message content/meaning.

Once this is merged, any translator gets a notification about a new text to translate.

My comment applies to **all** messages.json files

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: libs/angular/src/auth/components/two-factor-icon.component.ts (hunk 1 above)
Comment (quexten) [reply libs/angular/src/auth/components/two-factor-icon.component.ts]: Makes sense! Updated in 3ed56bc761.

File: apps/web/src/app/auth/settings/two-factor-authenticator.component.html [hunk 3]
```diff
@@ -5,65 +5,78 @@
...
-          <i class="bwi bwi-li bwi-android"></i>{{ "androidDevices" | i18n }}:
+          {{ "twoStepAuthenticatorInstructionInfix1" | i18n }}
           <a
             bitLink
-            href="https://play.google.com/store/apps/details?id=com.authy.authy"
             target="_blank"
             rel="noreferrer"
-            >Authy</a
+            (click)="launchExternalUrl('https://2fas.com')"
+            >2FAS Authenticator</a
```
[the remark this answers is not in the document]
Comment (quexten) [reply apps/web/src/app/auth/settings/two-factor-authenticator.component.html]: Updated in 9321068

File: apps/web/src/app/auth/settings/two-factor-authenticator.component.html [hunk 4]
```diff
@@ -5,65 +5,78 @@
...
+        <p>
+          {{ "twoStepAuthenticatorInstructionPrefix" | i18n }}
           <a
             bitLink
-            href="https://itunes.apple.com/us/app/authy/id494168017?mt=8"
             target="_blank"
             rel="noreferrer"
-            >Authy</a
+            (click)="launchExternalUrl('https://getaegis.app')"
+            >Aegis Authenticator</a
```
[the remark this answers is not in the document]
Comment (quexten) [reply apps/web/src/app/auth/settings/two-factor-authenticator.component.html]: Updated in 9321068

File: apps/web/src/locales/en/messages.json [hunk 5]
```diff
@@ -971,13 +971,13 @@
...
-    "message": "FIDO U2F security key"
+    "message": "FIDO U2F security a key"
   },
   "webAuthnTitle": {
-    "message": "FIDO2 WebAuthn"
+    "message": "Passkey"
   },
   "webAuthnDesc": {
-    "message": "Use any WebAuthn compatible security key to access your account."
+    "message": "Use your device's biometrics or a FIDO2 security key."
```
[the remark this answers is not in the document]
Comment (quexten) [reply apps/web/src/locales/en/messages.json]: Updated in d79c345. Nice catch!

File: apps/browser/src/_locales/en/messages.json (hunk 2 above)
Comment (quexten) [reply apps/browser/src/_locales/en/messages.json:1097]: To clarify, do I just rename the key in the English messages.json or do I do this for all languages?

Comment (quexten) [reply apps/browser/src/_locales/en/messages.json:1097]: Also is there a general pattern to prefer when simply updating the text to a different phrasing? Does adding a "V2" suffix to each key suffice?

Comment (quexten) [reply apps/browser/src/_locales/en/messages.json]: The V2 suffix is what I went with for now in a4ec25d , but I'm happy to change it if there is a preferable pattern.

Comment (djsmith85) [reply apps/browser/src/_locales/en/messages.json]: Only the keys in the en/messages.json need to change. Those are then used as source for all translations. 
Regarding naming, there are no guidelines. Usually an update in copy also introduces a different meaning which the key can be renamed to.

File: apps/web/src/locales/en/messages.json [hunk 6]
```diff
@@ -1626,35 +1626,44 @@
...
+  },
+  "twoStepAuthenticatorScanCodeV2": {
+    "message": "Scan the QR code below with your authenticator app or enter the key."
   },
   "key": {
     "message": "Key"
   },
-  "twoStepAuthenticatorEnterCode": {
-    "message": "Enter the resulting 6 digit verification code from the app"
+  "twoStepAuthenticatorEnterCode V2": {
```
Comment (djsmith85) [inline apps/web/src/locales/en/messages.json:1665]: Need to remove the space from the key, but other than that all changes are looking good.

Leaving it up to the Auth team to finish the review

**Review by djsmith85 — CHANGES_REQUESTED** [review]

File: apps/web/src/locales/en/messages.json (hunk 6 above)
Comment (quexten) [reply apps/web/src/locales/en/messages.json]: Messed this up in find&replace. Nice catch!

File: apps/browser/src/_locales/en/messages.json (hunk 2 above)
Comment (quexten) [reply apps/browser/src/_locales/en/messages.json]: Great! In this case most it's mostly the phrasing that changes, so keeping the V2 suffix.

**Comment by quexten** [comment]
@rr-bw Moved out the svg icons to separate files as per the example you linked.

**Review by djsmith85 — DISMISSED** [review]
@quexten Thank you for addressing my translation-related concerns!

**Review by rr-bw — APPROVED** [review]

**Comment by quexten** [comment]
@rr-bw Seems resolving merge conflicts dismissed the approving review.

**Review by rr-bw — APPROVED** [review]

**Pull request merged**